### ISSUE TITLE
Confusion between the Media and MediaSet tables

### ISSUE THREAD

**Issue by FSharpCSharp** [Microsoft/AL#474]
The handling of company-independent media tables is extremely difficult. The following errors occur:

`'The type or method 'Media' cannot be used for 'Extension' development.'`

`'The type or method 'Media Set' cannot be used for 'Extension' development.'`

No images can be used that are used in tables that are independent of a company.

**Comment by esbenk** [comment]
You should use the "Tenant Media" and "Tenant Media Set".

**Comment by FSharpCSharp** [comment]
This does not work for tables independent of the company. I can not find any references within this table if I have a Media or MediaSet field in a company-independent table.

How can you read the contents of the field otherwise?

**Comment by esbenk** [comment]
I've consulted with the developer of the feature. The tables are not intended to be used directly but should be accessed through the  ([API's](https://msdn.microsoft.com/en-us/dynamics-nav/developer/devenv-working-with-media-on-records)).

**Comment by FSharpCSharp** [comment]
OK, an API is here, of course, much more useful as a direct tabular access. This would also eliminate the confusion over the Media and MediaSet tables in the future. We have failed a bit with our first implementation, that one can read MediaSet only meaningfully via a direct table access. At that time, we also noticed that some tables are double (with and without tenant in the name). But in the future, then this is then hopefully simplified by the API!
